src/lib/storage/zero_suppression/simd_bp128/simd_bp128_decoder.cpp

```diff
@@ -0,0 +1,43 @@
+#include "simd_bp128_decoder.hpp"
+
+#include "simd_bp128_vector.hpp"
+
+namespace opossum {
+
+SimdBp128Decoder::SimdBp128Decoder(const SimdBp128Vector& vector)
+    : _data{&vector._data},
+      _size{vector._size},
+      _cached_meta_info_offset{0u},
+      _cached_meta_block_first_index{std::numeric_limits<size_t>::max()},
+      _cached_block_first_index{std::numeric_limits<size_t>::max()},
+      _cached_block{std::make_unique<std::array<uint32_t, Packing::block_size>>()} {}
+
+SimdBp128Decoder::SimdBp128Decoder(const SimdBp128Decoder& other)
+    : _data{other._data},
+      _size{other._size},
+      _cached_meta_info_offset{other._cached_meta_info_offset},
+      _cached_meta_block_first_index{other._cached_meta_block_first_index},
+      _cached_meta_info{other._cached_meta_info},
+      _cached_block_first_index{other._cached_block_first_index},
+      _cached_block{std::make_unique<std::array<uint32_t, Packing::block_size>>(*other._cached_block)} {}
+
+void SimdBp128Decoder::_read_meta_info(size_t meta_info_offset) {
+  Packing::read_meta_info(_data->data() + meta_info_offset, _cached_meta_info.data());
+}
+
+void SimdBp128Decoder::_unpack_block(uint8_t block_index) {
+  static const auto meta_info_data_size = 1u;  // One 128 bit block
```

Comment:
Shouldn't this magic number be somewhere central?

Reply:
It’s only used here. Should it?

Reply:
Maybe I'm not fully understanding the meta block concept yet. Could it also be two 128bit blocks?

Reply:
no.

Reply:
Ok